parquet-common/src/main/java/org/apache/parquet/CorruptStatistics.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet;
+
+import org.apache.parquet.VersionParser.ParsedVersion;
+
+/**
+ * There was a bug (PARQUET-251) that caused the statistics metadata
+ * for binary columns to be corrupted in the write path.
+ *
+ * This class is used to detect whether a file was written with this bug,
+ * and thus it's statistics should be ignored / not trusted.
+ */
+public class CorruptStatistics {
+  private static final Log LOG = Log.getLog(CorruptStatistics.class);
+
+  // the version in which the bug described by jira: PARQUET-251 was fixed
+  // the bug involved writing invalid binary statistics, so stats written prior to this
+  // fix must be ignored / assumed invalid
+  private static final SemanticVersion PARQUET_251_FIXED_VERSION = new SemanticVersion(1, 8, 0);
+
+  /**
+   * Decides if the statistics from a file created by createdBy (the created_by field from parquet format)
+   * should be ignored because they are potentially corrupt.
+   */
+  public static boolean shouldIgnoreStatistics(String createdBy) {
+
+    if (Strings.isNullOrEmpty(createdBy)) {
+      // created_by is not populated, which could have been caused by
+      // parquet-mr during the same time as PARQUET-251, see PARQUET-297
+      LOG.info("Ignoring statistics because created_by is null or empty! See PARQUET-251 and PARQUET-297");
+      return true;
+    }
+
+    try {
+      ParsedVersion version = VersionParser.parse(createdBy);
+
+      if (!"parquet-mr".equals(version.application)) {
+        // assume other applications don't have this bug
+        return false;
+      }
+
+      SemanticVersion semver = SemanticVersion.parse(version.semver);
+      if (semver.compareTo(PARQUET_251_FIXED_VERSION) < 0) {
+        LOG.info("Ignoring statistics because this file was created prior to "
+            + PARQUET_251_FIXED_VERSION
+            + ", see PARQUET-251" );
+        return true;
+      }
+
+      // this file was created after the fix
+      return false;
+    } catch (RuntimeException e) {
```

Comment:
This is the right way to handle the API defined by the version parsers and the test does verify that completely unknown strings are fine. But I think throwing a RuntimeException is going to cause problems later if any other code doesn't know to catch the exception, and that is likely to end up in not being able to read files.